# DAY 68 - Data Processing

1. The DataFrame is given below:

            amzn_open  amzn_high   amzn_low  amzn_close  amzn_volume  tsla_open  tsla_high  tsla_low  tsla_close  tsla_volume
Date                                                                                                                         
2020-09-22  3033.8350  3133.9900  3000.2000     3128.99      6948816     429.60     437.76  417.6001      424.23     79580795
2020-09-21  2906.5000  2962.0000  2871.0000     2960.47      6117925     453.13     455.68  407.0700      449.39    109476800
2020-09-18  3031.7400  3037.8000  2905.5400     2954.91      8892580     447.94     451.00  428.8000      442.15     86406819
2020-09-17  3009.2500  3029.4320  2972.5500     3008.73      6449050     415.60     437.79  408.0001      423.43     76779163
2020-09-16  3179.9900  3187.2380  3074.1500     3078.10      4531189     439.87     457.79  435.3100      441.76     72546760
2020-09-15  3136.1600  3175.0200  3108.9200     3156.13      4021535     436.56     461.94  430.7000      449.76     97298228
2020-09-14  3172.9372  3187.3900  3096.0000     3102.97      4529596     380.95     420.00  373.3000      419.62     83020608
2020-09-11  3208.6900  3217.3409  3083.9800     3116.22      5093982     381.94     382.50  360.5000      372.72     60717459
2020-09-10  3307.2188  3349.8900  3170.5500     3175.11      5330741     386.21     398.99  360.5600      371.34     84930608
2020-09-09  3202.9900  3303.1750  3185.0000     3268.61      5188675     356.60     369.00  341.5100      366.28     79465769
2020-09-08  3144.0000  3250.8450  3130.0000     3149.84      6094205     356.00     368.74  329.8800      330.21    115465691
2020-09-04  3318.0000  3381.5000  3111.1300     3294.62      8781754     402.81     428.00  372.0201      418.32    110321885
2020-09-03  3485.0000  3488.4100  3303.0000     3368.00      8161113     407.23     431.80  402.0000      407.00     87596086
2020-09-02  3547.0000  3552.2500  3486.6850     3531.45      3931476     478.99     479.04  405.1206      447.37     96176128
2020-09-01  3489.5800  3513.8700  3467.0000     3499.12      3476407     502.14     502.49  470.5100      475.05     90119419
2020-08-31  3408.9900  3495.0000  3405.0000     3450.96      4185885     444.61     500.14  440.1100      498.32    118374406
2020-08-28  3423.0000  3433.3699  3386.5000     3401.80      2896978     459.02     463.70  437.3000      442.68    100405880
2020-08-27  3450.0500  3453.0000  3378.0000     3400.00      4264795     436.09     459.12  428.5000      447.75    118465215
2020-08-26  3351.1100  3451.7387  3344.5675     3441.85      6508743     412.00     433.20  410.7300      430.63     71196910
2020-08-25  3294.9900  3357.4000  3267.0000     3346.49      3992842     394.98     405.59  393.6000      404.67     53294465


Extract the following columns from the result DataFrame:

amzn_open
amzn_close
tsla_open
tsla_close

and copy to a new object called quotations.

In [2]:
import pandas as pd
 
amazon = pd.read_csv('amazon.csv', index_col=0)
tesla = pd.read_csv('tesla.csv', index_col=0)
amazon.columns = ['amzn_' + col.lower() for col in amazon.columns]
tesla.columns = ['tsla_' + col.lower() for col in tesla.columns]
result = pd.concat([amazon, tesla], axis=1)
quotations = result[    ['amzn_open', 'amzn_close', 'tsla_open', 'tsla_close']  ].copy()
print(quotations)

            amzn_open  amzn_close  tsla_open  tsla_close
Date                                                    
2020-09-22  3033.8350     3128.99     429.60      424.23
2020-09-21  2906.5000     2960.47     453.13      449.39
2020-09-18  3031.7400     2954.91     447.94      442.15
2020-09-17  3009.2500     3008.73     415.60      423.43
2020-09-16  3179.9900     3078.10     439.87      441.76
2020-09-15  3136.1600     3156.13     436.56      449.76
2020-09-14  3172.9372     3102.97     380.95      419.62
2020-09-11  3208.6900     3116.22     381.94      372.72
2020-09-10  3307.2188     3175.11     386.21      371.34
2020-09-09  3202.9900     3268.61     356.60      366.28
2020-09-08  3144.0000     3149.84     356.00      330.21
2020-09-04  3318.0000     3294.62     402.81      418.32
2020-09-03  3485.0000     3368.00     407.23      407.00
2020-09-02  3547.0000     3531.45     478.99      447.37
2020-09-01  3489.5800     3499.12     502.14      475.05
2020-08-31  3408.9900     3450.

2. The quotations DataFrame is given below:

            amzn_open  amzn_close  tsla_open  tsla_close
Date                                                    
2020-09-22  3033.8350     3128.99     429.60      424.23
2020-09-21  2906.5000     2960.47     453.13      449.39
2020-09-18  3031.7400     2954.91     447.94      442.15
2020-09-17  3009.2500     3008.73     415.60      423.43
2020-09-16  3179.9900     3078.10     439.87      441.76
2020-09-15  3136.1600     3156.13     436.56      449.76
2020-09-14  3172.9372     3102.97     380.95      419.62
2020-09-11  3208.6900     3116.22     381.94      372.72
2020-09-10  3307.2188     3175.11     386.21      371.34
2020-09-09  3202.9900     3268.61     356.60      366.28
2020-09-08  3144.0000     3149.84     356.00      330.21
2020-09-04  3318.0000     3294.62     402.81      418.32
2020-09-03  3485.0000     3368.00     407.23      407.00
2020-09-02  3547.0000     3531.45     478.99      447.37
2020-09-01  3489.5800     3499.12     502.14      475.05
2020-08-31  3408.9900     3450.96     444.61      498.32
2020-08-28  3423.0000     3401.80     459.02      442.68
2020-08-27  3450.0500     3400.00     436.09      447.75
2020-08-26  3351.1100     3441.85     412.00      430.63
2020-08-25  3294.9900     3346.49     394.98      404.67

Assign two following columns to the quotations DataFrame:

amzn_change
tsla_change

which will be the daily percentage change of the price.

In [3]:
import pandas as pd
 
amazon = pd.read_csv('amazon.csv', index_col=0)
tesla = pd.read_csv('tesla.csv', index_col=0)
amazon.columns = ['amzn_' + col.lower() for col in amazon.columns]
tesla.columns = ['tsla_' + col.lower() for col in tesla.columns]
result = pd.concat([amazon, tesla], axis=1)
quotations = result[    ['amzn_open', 'amzn_close', 'tsla_open', 'tsla_close']  ].copy()
quotations['amzn_change'] = (quotations['amzn_close'] / quotations['amzn_open'] - 1 ) * 100
quotations['tsla_change'] = ( quotations['tsla_close'] / quotations['tsla_open'] - 1 ) * 100
print(quotations)

            amzn_open  amzn_close  tsla_open  tsla_close  amzn_change  \
Date                                                                    
2020-09-22  3033.8350     3128.99     429.60      424.23     3.136459   
2020-09-21  2906.5000     2960.47     453.13      449.39     1.856873   
2020-09-18  3031.7400     2954.91     447.94      442.15    -2.534188   
2020-09-17  3009.2500     3008.73     415.60      423.43    -0.017280   
2020-09-16  3179.9900     3078.10     439.87      441.76    -3.204098   
2020-09-15  3136.1600     3156.13     436.56      449.76     0.636766   
2020-09-14  3172.9372     3102.97     380.95      419.62    -2.205124   
2020-09-11  3208.6900     3116.22     381.94      372.72    -2.881861   
2020-09-10  3307.2188     3175.11     386.21      371.34    -3.994559   
2020-09-09  3202.9900     3268.61     356.60      366.28     2.048711   
2020-09-08  3144.0000     3149.84     356.00      330.21     0.185751   
2020-09-04  3318.0000     3294.62     402.81      4

3. The quotations DataFrame is given below:

            amzn_open  amzn_close  tsla_open  tsla_close  amzn_change  tsla_change
Date                                                                              
2020-09-22  3033.8350     3128.99     429.60      424.23     3.136459    -1.250000
2020-09-21  2906.5000     2960.47     453.13      449.39     1.856873    -0.825370
2020-09-18  3031.7400     2954.91     447.94      442.15    -2.534188    -1.292584
2020-09-17  3009.2500     3008.73     415.60      423.43    -0.017280     1.884023
2020-09-16  3179.9900     3078.10     439.87      441.76    -3.204098     0.429672
2020-09-15  3136.1600     3156.13     436.56      449.76     0.636766     3.023639
2020-09-14  3172.9372     3102.97     380.95      419.62    -2.205124    10.150938
2020-09-11  3208.6900     3116.22     381.94      372.72    -2.881861    -2.413992
2020-09-10  3307.2188     3175.11     386.21      371.34    -3.994559    -3.850237
2020-09-09  3202.9900     3268.61     356.60      366.28     2.048711     2.714526
2020-09-08  3144.0000     3149.84     356.00      330.21     0.185751    -7.244382
2020-09-04  3318.0000     3294.62     402.81      418.32    -0.704641     3.850451
2020-09-03  3485.0000     3368.00     407.23      407.00    -3.357245    -0.056479
2020-09-02  3547.0000     3531.45     478.99      447.37    -0.438399    -6.601390
2020-09-01  3489.5800     3499.12     502.14      475.05     0.273385    -5.394910
2020-08-31  3408.9900     3450.96     444.61      498.32     1.231156    12.080250
2020-08-28  3423.0000     3401.80     459.02      442.68    -0.619340    -3.559758
2020-08-27  3450.0500     3400.00     436.09      447.75    -1.450704     2.673760
2020-08-26  3351.1100     3441.85     412.00      430.63     2.707760     4.521845
2020-08-25  3294.9900     3346.49     394.98      404.67     1.562979     2.453289

Create a new column 'similarity_flag' that stores 1 when the daily percentage change for Amazon and Tesla is of the same sign, otherwise 0 as shown below.

In [4]:
import pandas as pd

amazon = pd.read_csv('amazon.csv', index_col=0)
tesla = pd.read_csv('tesla.csv', index_col=0)
amazon.columns = ['amzn_' + col.lower() for col in amazon.columns]
tesla.columns = ['tsla_' + col.lower() for col in tesla.columns]
result = pd.concat([amazon, tesla], axis=1)
quotations = result[    ['amzn_open', 'amzn_close', 'tsla_open', 'tsla_close']  ].copy()
quotations['amzn_change'] = ( quotations['amzn_close'] / quotations['amzn_open'] - 1 ) * 100
quotations['tsla_change'] = ( quotations['tsla_close'] / quotations['tsla_open'] - 1 ) * 100
quotations['similarity_flag'] = ( quotations['amzn_change'] * quotations['tsla_change'] > 0 ) * 1
print(quotations)

            amzn_open  amzn_close  tsla_open  tsla_close  amzn_change  \
Date                                                                    
2020-09-22  3033.8350     3128.99     429.60      424.23     3.136459   
2020-09-21  2906.5000     2960.47     453.13      449.39     1.856873   
2020-09-18  3031.7400     2954.91     447.94      442.15    -2.534188   
2020-09-17  3009.2500     3008.73     415.60      423.43    -0.017280   
2020-09-16  3179.9900     3078.10     439.87      441.76    -3.204098   
2020-09-15  3136.1600     3156.13     436.56      449.76     0.636766   
2020-09-14  3172.9372     3102.97     380.95      419.62    -2.205124   
2020-09-11  3208.6900     3116.22     381.94      372.72    -2.881861   
2020-09-10  3307.2188     3175.11     386.21      371.34    -3.994559   
2020-09-09  3202.9900     3268.61     356.60      366.28     2.048711   
2020-09-08  3144.0000     3149.84     356.00      330.21     0.185751   
2020-09-04  3318.0000     3294.62     402.81      4